tests/test_utils.py

```diff
@@ -27,3 +28,118 @@ def bla(_result_buffer=None, result_dtype=None):
     got = bla(result_dtype=np.float)
     np.testing.assert_equal(got, should_get)
     assert got.dtype == should_get.dtype
+
+
+@hst.composite
+def get_dummy_data(draw,
+                   data_length=(0, 10),
+                   dt=(1, 10),
+                   max_time=(1, 20)
+                   ):
+    """
+    Create some dummy array for testing apply selection. Data may be
+        overlapping.
+
+    :param data_length: desired length of the data
+    :param dt: duration of each sample
+    :return: data
+    """
+    # Convert ranges to int
+    data_length = draw(hst.integers(*data_length))
+    dt = draw(hst.integers(*dt))
+    max_time = draw(hst.integers(*max_time))
+
+    data = np.zeros(data_length, dtype=strax.time_fields + [('data', np.float)])
+    data['time'] = np.random.randint(0, max_time + 1, data_length)
```

Comment:
You could also draw an array of times https://hypothesis.readthedocs.io/en/latest/numpy.html#numpy then it would be tracked by hypothesis too.

Reply:
Thanks Daniel, I think the input array is now already tracked when failing the tests. Will look at such feature next time I write a similar test.

Reply:
Okay as you like. I do not know if hypothesis will track this if an exmapel fails.

Reply:
It does, it's very easy to test right:
```python
================================================= test session starts ==================================================platform linux -- Python 3.8.3, pytest-6.2.0, py-1.9.0, pluggy-0.13.1
rootdir: /mnt/d/Google_Drive/PhD-master/ubuntu-storage/ubuntu-windows/software/strax_official
plugins: hypothesis-5.43.3, cov-2.11.1
collected 4 items

tests/test_utils.py .F..                                                                                         [100%]

======================================================= FAILURES =======================================================_________________________________________________ test_time_selection __________________________________________________
>   ???

tests/test_utils.py:61:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

d = array([],
      dtype=[(('Start time since unix epoch [ns]', 'time'), '<i8'), (('Exclusive end time since unix epoch [ns]', 'endtime'), '<i8'), ('data', '<f8')])
second_time = 0, second_dt = 1

    @settings(deadline=None)
    @given(get_dummy_data(),
           hst.integers(0, 20),
           hst.integers(1, 10),
           )
    def test_time_selection(d, second_time, second_dt):
        """
        Test that both 'touching' and 'fully_contained' give the same
            results as 'strax.fully_contained_in' and
            'strax.touching_windows' respectively
        :param d: test-data from get_dummy_data
        :param second_dt: the ofset w.r.t. the first
        :return: None
        """
>       assert False, d
E       AssertionError: array([],
E               dtype=[(('Start time since unix epoch [ns]', 'time'), '<i8'), (('Exclusive end time since unix epoch [ns]', 'endtime'), '<i8'), ('data', '<f8')])
E       assert False

tests/test_utils.py:74: AssertionError
------------------------------------------------------ Hypothesis ------------------------------------------------------Falsifying example: test_time_selection(
    d=array([],
          dtype=[(('Start time since unix epoch [ns]', 'time'), '<i8'), (('Exclusive end time since unix epoch [ns]', 'endtime'), '<i8'), ('data', '<f8')]),
    second_time=0,
    second_dt=1,
)
=============================================== short test summary info ================================================FAILED tests/test_utils.py::test_time_selection - AssertionError: array([],
```

Reply:
No I meant internally in its example cache, to reproduce the exact same condition next time you make a PR.

Reply:
I see, in principle you can set this as an example, which makes it fully reproducible:
```python
    d=array([],
          dtype=[(('Start time since unix epoch [ns]', 'time'), '<i8'), (('Exclusive end time since unix epoch [ns]', 'endtime'), '<i8'), ('data', '<f8')]),
    second_time=0,
    second_dt=1,
```

Will take it more into account next time writing such a test. Thanks for showing it! 
(This specific one is very unlikely to fail, especially in the parts where this reproducibility is an issue but it's very much appreciated that you correctly pointed this out).